### DISCUSSION TITLE
Package Vulnerability: Dependency Confusion Attack - uWebSockets.js

### DISCUSSION THREAD

**Discussion by SwhiteMHC** [enisdenjo/graphql-ws#596] [Q&A] [answered]
uWebSockets.js is not an npm based package and is vulnerable to a dependency confusion attack. This is being flagged during security audits.

https://unetworkingab.medium.com/beware-of-tin-foil-hattery-f738b620468c This article has some back story.

**Comment by enisdenjo** [comment, marked as answer]
I know about that old article and the uWS author explains there why NPM is not for him. I respect his choice.

Furthermore, uWS is an optional peer dependency and as such does not pose a security threat from within the package.
